### PR TITLE
trace: add force_trace_http sample

### PR DESCRIPTION
**Pull request by broady**

### CODE REVIEW

**Comment by broady** [comment]
@rakyll PTAL

**Comment by broady** [comment]
@derekperkins FYI

File: trace/force_trace_http/main.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		log.Fatalf("Could not start Cloud Trace: %v", err)
+	}
+
+	http.HandleFunc("/", slowHandler)
+	// Force tracing of every request.
+	http.ListenAndServe(":8080", http.HandlerFunc(forceTraceHandler))
+}
+
+func forceTraceHandler(w http.ResponseWriter, r *http.Request) {
+	if h := "X-Cloud-Trace-Context"; r.Header.Get(h) == "" {
```
Comment (rakyll) [inline trace/force_trace_http/main.go]: Maybe put it here as a constant

const h = "X-Cloud-Trace-Context"

File: trace/force_trace_http/main.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		// Generate a trace header.
+		// https://cloud.google.com/trace/docs/faq
+		r.Header.Set(h, generateTraceID()+"/0;o=1")
+	}
+	http.DefaultServeMux.ServeHTTP(w, r)
+}
+
+func generateTraceID() string {
+	b := make([]byte, 16)
+	rand.Read(b)
```
Comment (rakyll) [inline trace/force_trace_http/main.go]: Handle the error returned from rand.Read.

File: trace/force_trace_http/main.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	return fmt.Sprintf("%x", b)
+}
+
+func slowHandler(w http.ResponseWriter, r *http.Request) {
+	span := traceClient.SpanFromRequest(r)
+	defer span.Finish()
+
+	time.Sleep(50 * time.Millisecond)
+	{
```
Comment (rakyll) [inline trace/force_trace_http/main.go]: Interesting style :)

**Comment by rakyll** [comment]
Sorry for the late feedback, a few nits, otherwise LGTM.

**Comment by rakyll** [comment]
With the existence of (*trace.Client).NewSpan now, we shouldn't merge this sample.

For cases users might want to trace regardless of the parent, they should depend on (*trace.Client).NewSpan instead.

**Pull request closed without merging**